packages/react/src/runtime/ac.ts

```diff
@@ -0,0 +1,131 @@
+const UNDERSCORE_UNICODE = 95;
+
+/**
+ * This length includes the underscore,
+ * e.g. `"_1s4A"` would be a valid atomic group hash.
+ */
+const ATOMIC_GROUP_LENGTH = 5;
+
+/**
+ * Memoize the result of ac so if it is called with the same args, it returns immediately.
+ * Also, to prevent useless React rerenders
+ */
+const cache = new Map();
```

Comment:
Did you look into the memory footprint and whether it was worth using a LRU cache? https://en.wikipedia.org/wiki/Cache_replacement_policies#Least_recently_used_(LRU)

Reply:
Initially I thought it is fine because `style-cache` uses [a same pattern](https://github.com/atlassian-labs/compiled/blob/master/packages/react/src/runtime/style-cache.tsx#L12). But you've raised a very good point. Let me look into it futher.

Reply:
> Did you look into the memory footprint and whether it was worth using a LRU cache? 

Thanks for pointing it out. I've tested it on my staging instance. The memory size of cache on issue view is ~50kb. It grows up to 100kb after clicking around and jumping between pages. I am leaning towards **not** implement LRU because
- The cache shouldn't grow indefinitely even if the page stays open for a long time as there are limited number of Compiled components
- LRU increases runtime cost

What do you think? @JakeLane @dddlr

Reply:
> also are there any implications for the bundler when using a side-effect singleton to store the cache? Do we need to initialise it somewhere to prevent multiple cache instances and breaking tree-shaking?

I've implemented https://github.com/atlassian-labs/compiled/pull/1437/files#diff-33bd4912ff7f9f36d0dd9ce38ba892dd1407cb49bb890319400819bd70c12047 to ensure only one runtime exist in the bundle.

Reply:
> Thanks for pointing it out. I've tested it on my staging instance. The memory size of cache on issue view is ~50kb. It grows up to 100kb after clicking around and jumping between pages.

Thank you for looking into it and for your insight @liamqma! Just to confirm, would you anticipate that the cache won't grow much beyond ~100 kB even in a (worst-case?) scenario where the user interacts with many different pages over a long span of time?

Also interested to hear what Jake thinks